File: pc-editor/src/main/java/com/auto_persona/pc/model/SaveData.java
package com.auto_persona.pc.model;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class SaveData {
    public String version = "1.0.0";
    public long timestamp;
    public String exportDate = "";
    public int slotId = 1;
    public GameContainer data = new GameContainer();

    public static class GameContainer {
        public GameData gameData = new GameData();
        public List<DiaryEntry> diary = List.of();
        public AppSettings settings = new AppSettings();
        public PromptsMap prompts = new PromptsMap();
    }
}

File: app/src/main/java/com/example/auto_persona/data/model/SaveData.kt
package com.example.auto_persona.data.model

import kotlinx.serialization.Serializable

@Serializable
data class SaveData(
    val version: String = "1.0.0",
    val timestamp: Long = 0L,
    val exportDate: String = "",
    val slotId: Int = 1,
    val data: GameContainer = GameContainer()
)

@Serializable
data class GameContainer(
    val gameData: GameData = GameData(),
    val diary: List<DiaryEntry> = emptyList(),
    val settings: AppSettings = AppSettings(),
    val prompts: PromptsMap = PromptsMap()
)
